### DISCUSSION TITLE
How to rerender after changing gltf scale?

### DISCUSSION THREAD

**Discussion by naonvl** [google/model-viewer#3502] [Q&A] [answered]
I'm trying to create product customizer which user can change similar to this https://launch.playcanvas.com/1430353?debug=true, that was mine actually but i wanted to move to model viewer since it's really time saving.

I was trying to change part of the gltf like this,
```
let root: any = Object.getOwnPropertySymbols(this.model.nativeElement.model).find(x => x.description === "roots");
this.model.nativeElement.model[root][1].mesh.scale.set(0, 0, 0);
```

but it didn't change on the viewer, but after i move the camera then it's started to change.
after clicking change button
![image](https://user-images.githubusercontent.com/43715306/170923246-06cc423f-6fb7-40d0-ae04-a86fc10b7bdb.png)
after moving the camera
![image](https://user-images.githubusercontent.com/43715306/170923301-1b1bd39b-eecb-4832-9b95-4d58c37bc146.png)

how to render the changes?

**Comment by naonvl** [comment, marked as answer]
nvm i use `interact()` as a trick like so
```
    const finger1 = {
      x: {
        initialValue: 0.4,
        keyframes: [
          { frames: 1, value: 0.3 },
        ]
      },
      y: {
        initialValue: 0.4,
        keyframes: [
          { frames: 1, value: 0.3 },
        ]
      }
    };
    this.model.nativeElement.interact(100,finger1,finger1);
```

so now the changes is updated

**Reply by elalish** [reply]
Since you're outside of our public API anyway with `getOwnSymbols`, you might just look for the `[$needsRender]()` function.

**Reply by naonvl** [reply]
thanks for the reply, how to access this `[$needsRender]()` btw?

**Reply by elalish** [reply]
`getOwnPropertySymbols` will let you hunt for any of our non-public functions; they all start with `$`.

**Reply by Frank3K** [reply]
I could not find the method, but found this to work:

```ts
this.modelViewer[Object.getOwnPropertySymbols(this.modelViewer).find(x => x.description === 'scene')].queueRender()
```

Source: https://github.com/google/model-viewer/blob/d10c010fd66817ad1bdb448c24de3ccc2db31c8e/packages/model-viewer/src/model-viewer-base.ts#L560-L562
